### DISCUSSION TITLE
[@types/three] ColorRepresentation is now overwriten with array of numbers

### DISCUSSION THREAD

**Discussion by aljaz-zupan** [DefinitelyTyped/DefinitelyTyped#66812] [Issues with a @types package]
I am using Threlte which is a wrapper for ThreeJs. Since version 0.153 of @types/three I can not pass the color as a string without typescript showing an error (passing strings works, error is for wrong param type). So if I use "red" or "#f3eeee" this was not showing as an error till 0.152.1.
![Screenshot from 2023-09-24 14-52-01](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/7879087/d14d1ddc-0dec-4a19-b57f-c496991a4d01)

Type was "ColorRepresentation" now shows as an array of numbers.

Somehow the set is being overwritten with numerical [(merge).](https://github.com/DefinitelyTyped/DefinitelyTyped/commit/b306148f699d677d64a633593f428f55a49d5e5e) See attached image.
![Screenshot from 2023-09-24 09-10-33](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/7879087/587d6f94-c60d-45c6-b717-18b9ea0f804a)

Edit: typos and clarity

[a comment by an automated account is not shown]

**Comment by Methuselah96** [comment]
Hey @aljaz-zupan, thanks for the issue!

I haven't found where in your types the `T.MeshStandardMaterial` type is being declared, but I assume it's using the `Paramaters` type utility to determine the types for the `color` prop.

Unfortunately, the `Parameters` type [only finds the parameters for the last signature of an overloaded function](https://www.typescriptlang.org/docs/handbook/utility-types.html#parameterstype), which is why issue is occurring. `@react-three/fiber` also ran into this issue and worked around it in https://github.com/pmndrs/react-three-fiber/pull/2931.

What type(s) are you expecting the `color` prop to expect in your case? Should you only be able to pass in a `ColorRepresentation` or should you be able to pass in an array of numbers as well? If the former, maybe we could reorder the overloads to make your life easier. If you would like to do this, please create a PR at https://github.com/three-types/three-ts-types.

Otherwise you may have to work around it like `@react-three/fiber` did. This seems like an unfortunate limitation of TypeScript.

**Comment by aljaz-zupan** [comment]
There is a fix being merged for this type of issue. [merge](https://github.com/three-types/three-ts-types/pull/614). Thank you for your help and time its much appreciated.
